src/NuGet.Core/NuGet.Resolver/PackageResolver.cs

```diff
@@ -162,6 +187,31 @@ public IEnumerable<PackageIdentity> Resolve(PackageResolverContext context, Canc
             throw new NuGetResolverConstraintException(message);
         }
 
+        private static IEnumerable<PackageDependency> GetBrokenDependencies(SourcePackageDependencyInfo package, IEnumerable<PackageIdentity> packages)
+        {
+            foreach (var dependency in package.Dependencies)
+            {
+                var target = packages.FirstOrDefault(targetPackage => StringComparer.OrdinalIgnoreCase.Equals(targetPackage.Id, dependency.Id));
+
+                if (!ResolverUtility.IsDependencySatisfied(dependency, target))
+                {
+                    yield return dependency;
+                }
+            }
+
+            yield break;
+        }
+
+        private static string FormatDependencyConstraint(SourcePackageDependencyInfo package, PackageDependency dependency)
+        {
+            var range = dependency.VersionRange;
+            var dependencyString = String.Format(CultureInfo.InvariantCulture, "{0} {1}", dependency.Id,
+                range == null ? string.Empty : range.ToNonSnapshotRange().PrettyPrint()).Trim();
+
+            // A 1.0.0 dependency: B (= 1.5)
+            return $"'{package.Id} {package.Version.ToNormalizedString()} {Strings.DependencyConstraint}: {dependencyString}'";
```

Comment:
make sure the versions gets normalized or not the same way. It should follow the pattern we use everywhere else in logging